bson/primitive/decimal.go

```diff
@@ -211,6 +212,22 @@ func (d Decimal128) IsZero() bool {
 	return d.h == 0 && d.l == 0
 }
 
+// MarshalJSON returns Decimal128 as a string.
```

Comment:
Perhaps best discussed offline, but I have a high level question.

Is the main use case for marshaling Decimal128 into a string (as opposed to something else) for readability?

Could we encode it as the canonical extended JSON form: `{"$numberDecimal": "1.23"}`? It's more verbose, but also less ambiguous.

It seems like this is consistent with other BSON types (e.g. ObjectID marshals into a string).

Reply:
Not sure, but I think we discussed this offline? But the idea is users don't expect the extended json syntax, and we want to keep consistency.

Reply:
Yes, from what I gathered we don't recommend users marshal directly to JSON instead of extended JSON, but they do and expect it to work.